core/src/replay_stage.rs

```diff
@@ -3239,13 +3244,30 @@ impl ReplayStage {
                 .expect("Entry in progress map must exist in BankForks")
                 .clone();
 
-            did_newly_reach_threshold = Self::update_slot_propagated_threshold_from_votes(
+            let propagation_updated = Self::update_slot_propagated_threshold_from_votes(
                 &mut newly_voted_pubkeys,
                 &mut cluster_slot_pubkeys,
                 &leader_bank,
                 leader_propagated_stats,
                 did_newly_reach_threshold,
-            ) || did_newly_reach_threshold;
+            );
+            if propagation_updated {
+                if let Some(info) = prior_retransmit_info {
+                    datapoint_info!(
+                        "replay_stage-retransmitted-propagated",
+                        ("slot", current_leader_slot.unwrap(), i64),
+                        ("retry_iteration", info.retry_iteration, i64),
+                        (
+                            "last_retry_elapsed_ms",
+                            info.retry_time
+                                .map(|t| t.elapsed().as_millis())
+                                .unwrap_or_default(),
+                            i64
+                        ),
+                    );
+                }
+            }
```

Comment:
How are the retried slots which don't reach threshold counted here?
It seems like it is only counting the ones which reach the threshold.

Also, I am not sure we really care about slot getting propagated or not.
What matters at the end is if the slot got rooted or not. If it didn't get rooted, then any effort to retry and propagate the slot was just wasting bandwidth and other resources; even if those retries achieve propagation.

To track what percentage of retried slots got rooted at the end, we already have a metric which tracks when the slot gets retried:
https://github.com/solana-labs/solana/blob/b77e12b3a/core/src/replay_stage.rs#L1068-L1073

We also know which slots get rooted either from the ledger or other metrics.
So we should be able to easily compute % of retried slots which got rooted at the end.